File: demos/php/dsa/trees/heap.php
<?php
echo "=== PHP SplPriorityQueue (Max Heap) ===\n";
$pq = new SplPriorityQueue();
$pq->insert("Task Low", 10);
$pq->insert("Task High", 50);
$pq->insert("Task Mid", 30);

assert($pq->top() === "Task High");
assert($pq->extract() === "Task High");
assert($pq->extract() === "Task Mid");
assert($pq->extract() === "Task Low");

echo "PHP SplPriorityQueue tests passed successfully.\n";

File: demos/rust/dsa/trees/heap.rs
use std::collections::BinaryHeap;

fn main() {
    println!("=== Rust BinaryHeap (Priority Queue) ===");
    let mut heap = BinaryHeap::new();
    heap.push(15);
    heap.push(50);
    heap.push(30);

    assert_eq!(heap.peek(), Some(&50));
    assert_eq!(heap.pop(), Some(50));
    assert_eq!(heap.pop(), Some(30));
    assert_eq!(heap.pop(), Some(15));
    assert_eq!(heap.pop(), None);

    println!("Rust BinaryHeap tests passed successfully.");
}
